src/objects/domain.ts

```diff
@@ -1,37 +1,66 @@
+import isEqual from 'lodash.isequal'
+import { watch } from 'vue'
 import { waitForConfig } from '@/objects/config'
 import { EventKey, emitter, waitForEvent } from '@/objects/events'
+import { ScaleXDomain, ScaleYDomain } from '@/objects/scales'
+import { waitForScope } from '@/objects/scope'
+import { waitForStage } from '@/objects/stage'
 import { graphDataFactory } from '@/utilities/graphDataFactory'
 
-export type RunGraphDomain = [start: Date, end: Date]
+export type RunGraphDomain = {
+  x: ScaleXDomain,
+  y: ScaleYDomain,
+}
 
 const { fetch: getData, stop: stopData } = graphDataFactory()
 
+let domainX: ScaleXDomain | null = null
+let domainY: ScaleYDomain | null = null
 let domain: RunGraphDomain | null = null
 
-export async function startDomain(): Promise<void> {
-  const config = await waitForConfig()
+export function startDomain(domain: () => RunGraphDomain | undefined): void {
+  const value = domain()
 
-  getGraphData(config.runId)
+  if (value) {
+    setDomain(value)
+  }
 
-  emitter.on('configUpdated', config => getGraphData(config.runId))
+  watchDomain(domain)
+  startDomainX()
+  startDomainY()
 }
 
 export function stopDomain(): void {
+  domainX = null
+  domainY = null
   domain = null
   stopData()
 }
 
-function getGraphData(runId: string): void {
-  getData(runId, ({ start_time, end_time }) => {
-    const event: EventKey = domain ? 'domainUpdated' : 'domainCreated'
+export function setDomain(value: Partial<RunGraphDomain>): void {
+  if (isEqual(domain, value)) {
+    return
```

Comment:
I think you might still get unwanted events if the caller passes a partial update that matches the existing domain, like:
```
const domain = { x: 1, y: 2 }

setDomain({ x: 1 })
```
Since `{ x: 1, y: 2} !== { x: 1 }` as this check is comparing

Reply:
Ah good catch. I should check x and y independently

Reply:
Updated 👍